core/templates/services/insert-script.service.ts

```diff
@@ -22,41 +22,69 @@ import {downgradeInjectable} from '@angular/upgrade/static';
 export enum KNOWN_SCRIPTS {
   DONORBOX = 'DONORBOX',
   UNKNOWN = 'UNKNOWN',
+  MATHJAX = 'MATHJAX',
 }
 
 @Injectable({
   providedIn: 'root',
 })
 export class InsertScriptService {
   private loadedScripts: Set<string> = new Set<string>();
+  private scriptsLoading: Map<string, Promise<void>> = new Map();
```

Comment:
Can you add comments to explain what each of these variables represents? I don't understand the difference between loadedScripts and scriptsLoading.

Also I think I'm a bit confused about why you need the scriptsLoading variable at all. Are we trying to guard against loading the same script twice, or is it intended to be read by something else (e.g. tests)? I guess I don't understand what would fail if you deleted all code references to scriptsLoading below...

Reply:
scriptsLoading is indeed used to prevent loading the same script twice. Would it be more helpful if we renamed it to scriptsCurrentlyLoading? I'll also add comments explaining what the variables do

Reply:
Ah ok, thanks, I think comments would help. I think a general comment that explains that this is to avoid loading the same script twice would be helpful.

Re naming, perhaps fullyLoadedScripts and partiallyLoadedScripts?

Some other questions (maybe you can answer them in the code comments):
- Why would the same script load twice anyway?
- What specific scripts does this occur for?